src/App.test.js

```diff
@@ -1,18 +1,26 @@
-import React from 'react';
-import { render } from '@testing-library/react';
-import { Provider } from 'react-redux';
-import store from './app/store';
-import App from './App';
+import React from "react";
+import { render,waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
 
-test('renders learn react link', async () => {
- const {queryAllByTestId}= render(
+import { Provider } from "react-redux";
+import store from "./app/store";
+import App from "./App";
+
+test("renders learn react link",  async () => {
+  const { queryAllByTestId, queryByTestId } = render(
     <Provider store={store}>
       <App />
     </Provider>
   );
 
-  const inputs = queryAllByTestId('inputField');
-
-  expect(inputs.length).toBe(2);
+  const input = queryByTestId("perPageSelector");
   
+  expect(queryAllByTestId("post").length).toBe(0);
+  userEvent.type(input, "3");
+  
+  await waitFor(() => {
+    expect(input).toHaveValue(3);

```

Comment:
someone move this line outside the `waitFor` block, move it one line up, it does have to be in the waitFor as it happen sync and only the state change is the part that happens async